examples/example-phx-1_2/config/config.exs

```diff
@@ -26,7 +26,8 @@ config :logger, :console,
 config :thesis,
   store: Thesis.EctoStore,
   authorization: Example.ThesisAuth,
-  uploader: Thesis.RepoUploader
+  uploader: Thesis.RepoUploader,
+  enable_cache: true
```

Comment:
I'd like to allow them to specify their own cache store if they want, rather than just a boolean true/false.

To do that, let's change this to `cache: Thesis.CacheStore`. We can still allow them to specify a `cache_name` if they want.

Reply:
Yeah, I like your solution a lot better :) The one thing I'm not crazy about though is the `cache_name` option - that's really a specific sub-setting. What about if we could do something like:

```elixir
config :thesis, cache: {Thesis.CacheStore, [other: :options, can: "go here"]}
```

And then our store behavior functions could (optionally with a [] default) accept those options as a second argument like so:

```elixir
def page(slug, [cache_name: name]) do
  ConCache.get_or_store(name, slug, fn() ->
    store().page(slug)
  end)
end

def page_contents(slug, opts) when is_binary(slug) do
  page_contents(page(slug, opts))
end
```

Users will be able to pass in configuration values from the thesis config to their own custom stores or caches without having to either hardcode it in their own modules or set up their own additional config.

Thoughts?

Reply:
You've got me thinking, @anthonylebrun ... I think to be consistent, we'd use this strategy instead (note the separate `Thesis.EctoStore` config just below the main config):

```elixir

# Configure thesis content editor
config :thesis,
  store: Thesis.EctoStore,
  cache: Thesis.CacheStore,
  authorization: Example.ThesisAuth,
  uploader: Thesis.RepoUploader

config :thesis, Thesis.EctoStore, repo: Example.Repo

config :thesis, Thesis.CacheStore,
  cache_name: "whut_whut",
  other_option: "heyo"
```

Reply:
Nice! This seems more idiomatic than what I was suggesting :) I'm still learning those Elixir conventions.